api/grpcserver.go

```diff
@@ -414,3 +423,71 @@ func (svr *GRPCServer) ReadContractStorage(ctx context.Context, in *iotexapi.Rea
 	}
 	return &iotexapi.ReadContractStorageResponse{Data: b}, nil
 }
+
+// TraceTransactionStructLogs get trace transaction struct logs
+func (svr *GRPCServer) TraceTransactionStructLogs(ctx context.Context, in *iotexapi.TraceTransactionStructLogsRequest) (*iotexapi.TraceTransactionStructLogsResponse, error) {
+	actInfo, err := svr.coreService.Action(util.Remove0xPrefix(in.GetActionHash()), false)
+	if err != nil {
+		return nil, err
+	}
+	exec, ok := actInfo.Action.Core.Action.(*iotextypes.ActionCore_Execution)
+	if !ok {
+		return nil, errUnsupportedAction
+	}
+
+	amount, _ := big.NewInt(0).SetString(exec.Execution.GetAmount(), 10)
+	callerAddr, err := address.FromString(actInfo.Sender)
+	if err != nil {
+		return nil, err
+	}
+	state, err := accountutil.AccountState(svr.coreService.sf, callerAddr)
+	if err != nil {
+		return nil, err
+	}
+	nonce := state.Nonce + 1
+	sc, _ := action.NewExecution(
+		exec.Execution.GetContract(),
+		nonce,
+		amount,
+		svr.coreService.cfg.Genesis.BlockGasLimit,
+		big.NewInt(0),
+		exec.Execution.GetData(),
+	)
+
+	ctx, err = svr.coreService.bc.Context(ctx)
+	if err != nil {
+		return nil, err
+	}
+	tracer := vm.NewStructLogger(nil)
+	vmConfig := vm.Config{Debug: true, Tracer: tracer, NoBaseFee: true}
+	ctx = protocol.WithVMConfigCtx(ctx, vmConfig)
+	_, _, err = svr.coreService.sf.SimulateExecution(ctx, callerAddr, sc, svr.coreService.dao.GetBlockHash)
+	if err != nil {
+		return nil, err
+	}
+
+	structLogs := make([]*iotextypes.TransactionStructLog, 0)
+	for _, log := range tracer.StructLogs() {
+		var stack []string
+		for _, s := range log.Stack {
+			stack = append(stack, s.String())
+		}
+		structLogs = append(structLogs, &iotextypes.TransactionStructLog{
+			Pc:         log.Pc,
+			Op:         uint64(log.Op),
+			Gas:        fmt.Sprintf("%#x", log.Gas),
+			GasCost:    fmt.Sprintf("%#x", log.GasCost),
+			Memory:     fmt.Sprintf("%#x", log.Memory),
```

Comment:
should we just print these as integer? for example, 10000 is more friendly vs. 0x2710?